github-connector/internal/registration/registerservice.go

```diff
@@ -1,1020 +1,123 @@
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/comments{/number}"
-											}
-										}
-									},
-									"commits": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/2/commits"
-											}
-										}
-									},
-									"statuses": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/statuses/ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-											}
-										}
-									}
-								}
-							},
-							"author_association": {
-								"type": "string",
-								"example": "OWNER"
-							}
-						}
-					},
-					"head": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:changes"
-							},
-							"ref": {
-								"type": "string",
-								"example": "changes"
-							},
-							"sha": {
-								"type": "string",
-								"example": "ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					},
-					"base": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:master"
-							},
-							"ref": {
-								"type": "string",
-								"example": "master"
-							},
-							"sha": {
-								"type": "string",
-								"example": "f95f852bd8fca8fcc58a9a2d6c842781e32a215e"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					}
-				}
-			}
-		}`),
-	},
+//Builder is an interface containing all necessary functions required to build an ServiceDetails structure
+type Builder interface {
+	BuildServiceDetails() (ServiceDetails, error)
+	GetApplicationRegistryURL() string
 }
 
-var url = "http://application-registry-external-api.kyma-integration.svc.cluster.local:8081/" + os.Getenv("GITHUB_CONNECTOR_NAME") + "-app/v1/metadata/services"
+//ServiceRegister is an interface containing all necessary functions required to register service in Kyma's application registry
+type ServiceRegister interface {
+	RegisterService() (string, apperrors.AppError)
+}
+
+type serviceRegister struct {
+	envName      string
+	builder      Builder
+	register     ServiceRegister
+	retryDelay   int
+	retriesCount int
+}
+
+//NewServiceRegister creates a serviceRegister instance with the passed in interface
+func NewServiceRegister(deploymentEnvName string, b Builder, retryTime int, retries int) serviceRegister {
+
```

Comment:
General hint:  what helps is checking how package name goes with component name and method names. Using the old names we have _registration.Builder_ (with _BuildServiceDetails_ method) and registration._ServiceRegister_ (with _RegisterService_ method). 
With new names we have _registration.PayloadBuilder_ (with _Build_ method) and _registration.ApplicationRegistryClient_ (with _RegisterService_ method).  
Package name gives a general context (_registration_) and component names (_ApplicationRegistryClient_ and _PayloadBuilder_) describe what they can do.

Reply:
Thanks for advice, this will definitely help us in the future. Changing it right now